pkg/k8s/service_cache.go

```diff
@@ -198,6 +198,26 @@ func (s *ServiceCache) UpdateService(k8sSvc *slim_corev1.Service, swg *lock.Stop
 	return svcID
 }
 
+func (s *ServiceCache) EnsureService(svcID ServiceID, swg *lock.StoppableWaitGroup) bool {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+	if svc, found := s.services[svcID]; found {
+		if endpoints, serviceReady := s.correlateEndpoints(svcID); serviceReady {
```

Comment:
Looks like you decided to go with option#1. Any reason why we can't just read the endpoints map [1] for a service to be restored?

[1] https://github.com/cilium/cilium/blob/master/pkg/k8s/service_cache.go#L91

Reply:
We can but then it's just a lot of duplicate logic in LRP manager to "construct" the LB.SVC structure to Upsert, it's much more straightforward to reuse the service event chan I think :)